pkg/ext/stores/tokens/tokens.go

```diff
@@ -1348,17 +1368,19 @@ func SessionID(ctx context.Context) (string, error) {
 	//   - multiple query attempts instead
 	userInfo, ok := request.UserFrom(ctx)
 	if !ok {
-		return "", fmt.Errorf("context has no provider/principal data")
+		return "", fmt.Errorf("context has no principal data")
 	}
 
 	extras := userInfo.GetExtra()
 	if extras == nil {
-		return "", fmt.Errorf("context has no provider/principal data")
+		return "", fmt.Errorf("context principal has no extras")
 	}
```

Comment:
This check is not needed. Chances of it being `nil` are slim to none, it's ok to read from the `nil` map.

Reply:
commit 50ba6fbc97f896c9899538e60ece54ccd5bbb6ae